File: solutions/julia/scrabble-score/2/scrabble-score.jl
scrabble_dict = 
    Dict(1=>['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T' ],
         2=>['D', 'G'],
         3=>['B', 'C', 'M', 'P' ],
         4=> ['F', 'H', 'V', 'W', 'Y'],  
         5=> ['K'], 
         8=> ['J','X'],
         10=> ['Q','Z'])

score(lt::AbstractChar) =
begin
    for score in keys(scrabble_dict) 
        if lt in scrabble_dict[score]
            return score
        end
    end
    return 0    
end

function score(str::String)
    str=="" && return 0
    reduce(+, map(score, collect(uppercase(str))))
 end

File: solutions/clojure/scrabble-score/2/src/scrabble_score.clj
(ns scrabble-score)



(defn score-letter [letter] ;; <- arglist goes here
      ;; your code goes here
 (let [l (clojure.string/lower-case letter)]
   (cond 
     (contains? #{"a" "e" "i" "o" "u" "l" "m" "n" "r" "s" "t"} l) 1
     (contains? #{"d" "g"} l) 2
     (contains? #{"b" "c" "m" "p"} l) 3
     (contains? #{"f" "h" "v" "w" "y"} l) 4
     (contains? #{"k"} l) 5
     (contains? #{"j" "x"} l) 8
     (contains? #{"q" "z"} l) 10
   ))
)

(defn score-word [word] ;; <- arglist goes here
  ;; your code goes here
  (reduce + (map score-letter word))
)
